internal/testcase/ecs_pod.go

```diff
@@ -1 +1,159 @@
+			p, err := pc.CreatePod(ctx, opts)
+			require.NoError(t, err)
+			require.NotZero(t, p)
+
+			err = p.Stop(ctx)
+			require.NoError(t, err)
+
+			info, err := p.Info(ctx)
+			require.NoError(t, err)
+			require.NotNil(t, info)
+
+			assert.Equal(t, cocoa.Stopped, info.Status)
+		},
+		"StopSucceedsWithSecrets": func(ctx context.Context, t *testing.T, v cocoa.Vault, pc cocoa.ECSPodCreator) {
+			p, err := pc.CreatePod(ctx, optsSecret)
+			require.NoError(t, err)
+			require.NotZero(t, p)
+
+			info, err := p.Info(ctx)
+			require.NoError(t, err)
+			require.NotNil(t, info)
+			require.NotNil(t, info.Resources)
+			assert.Equal(t, cocoa.Running, info.Status)
+			assert.Equal(t, 2, len(info.Resources.Secrets))
+
+			err = p.Stop(ctx)
+			require.NoError(t, err)
+
+			info, err = p.Info(ctx)
+			require.NoError(t, err)
+			require.NotNil(t, info)
+			assert.Equal(t, cocoa.Stopped, info.Status)
+			require.Equal(t, 2, len(info.Resources.Secrets))
+
+			arn := info.Resources.Secrets[0].Name
+			id, err := v.GetValue(ctx, *arn)
+			require.NoError(t, err)
+			require.NotNil(t, id)
+		},
+		"StopFailsOnIncorrectPodStatus": func(ctx context.Context, t *testing.T, v cocoa.Vault, pc cocoa.ECSPodCreator) {
+			p, err := pc.CreatePod(ctx, opts)
+			require.NoError(t, err)
+			require.NotZero(t, p)
+
+			err = p.Stop(ctx)
+			require.NoError(t, err)
+
+			info, err := p.Info(ctx)
+			require.NoError(t, err)
+			require.NotZero(t, info)
+			assert.Equal(t, cocoa.Stopped, info.Status)
+
+			err = p.Stop(ctx)
+			require.Error(t, err)
+		},
+		"DeleteSucceeds": func(ctx context.Context, t *testing.T, v cocoa.Vault, pc cocoa.ECSPodCreator) {
+			p, err := pc.CreatePod(ctx, opts)
+			require.NoError(t, err)
+			require.NotZero(t, p)
+
+			info, err := p.Info(ctx)
+			require.NoError(t, err)
+			require.NotNil(t, info)
+			assert.Equal(t, cocoa.Running, info.Status)
+
+			err = p.Delete(ctx)
+			require.NoError(t, err)
+
+			info, err = p.Info(ctx)
+			require.NoError(t, err)
+			require.NotNil(t, info)
+			assert.Equal(t, cocoa.Deleted, info.Status)
+		},
+		"DeleteSucceedsWithSecrets": func(ctx context.Context, t *testing.T, v cocoa.Vault, pc cocoa.ECSPodCreator) {
+			p, err := pc.CreatePod(ctx, optsSecret)
+			require.NoError(t, err)
+			require.NotZero(t, p)
+
+			info, err := p.Info(ctx)
+			require.NoError(t, err)
+			require.NotNil(t, info)
+			require.NotNil(t, info.Resources)
+			require.NotNil(t, info.Resources.Secrets)
+			assert.Equal(t, cocoa.Running, info.Status)
+			require.Equal(t, 2, len(info.Resources.Secrets))
+
+			err = p.Delete(ctx)
+			require.NoError(t, err)
+
+			info, err = p.Info(ctx)
+			require.NoError(t, err)
+			require.NotNil(t, info)
+			assert.Equal(t, cocoa.Deleted, info.Status)
+			require.Equal(t, 2, len(info.Resources.Secrets))
+
+			arn0 := info.Resources.Secrets[0].Name
+			id, err := v.GetValue(ctx, *arn0)
+			require.NoError(t, err)
+			require.NotZero(t, id)
```

Comment:
Is this supposed to check that the returned value equals the one you originally set?